core/src/main/scala/org/apache/spark/storage/StorageUtils.scala

```diff
@@ -118,8 +140,20 @@ class StorageStatus(val blockManagerId: BlockManagerId, val maxMem: Long) {
         } else {
           None
         }
+      case BroadcastBlockId(broadcastId, _) =>
+        // Actually remove the block, if it exists
+        if (_broadcastBlocks.contains(broadcastId)) {
+          val removed = _broadcastBlocks(broadcastId).remove(blockId)
+          // If the given RDD has no more blocks left, remove the RDD
```

Comment:
Please, don't do that. It is very hard to understand these kinds of extractors, especially when RddOrBlockId itself doesn't have a real class.